src/dags/signature.py

```diff
@@ -175,7 +193,7 @@ def rename_arguments(
 ) -> Callable[[Callable[P, R]], Callable[P, R]]: ...
 
 
-def rename_arguments(
+def rename_arguments(  # noqa: C901
     func: Callable[P, R] | None = None, *, mapper: dict[str, str] | None = None
```

Comment:
Why do we need None here for `func` ? Just wondering, no need to dig deeply into it.

Reply:
If we call the decorator only with the argument `mapper` it returns another decorator; otherwise, it returns the decorated function. Hence, this allows us to do the following:

```python
@rename_arguments(mapper=...)
def f(...):
   pass
```

instead of

```python
@functools.partial(rename_arguments, mapper=...)
def f(...):
   pass
```